### PR TITLE
Add Kudu bindings

### PR DESCRIPTION
**Pull request by jdcryans**
Currently doesn't include unit tests since we don't have mock objects for that.

### CODE REVIEW

File: kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+
+    client = new KuduClient.KuduClientBuilder(masterQuorum)
+        .defaultSocketReadTimeoutMs(DEFAULT_SLEEP)
+        .defaultOperationTimeoutMs(DEFAULT_SLEEP)
+        .build();
+    if (debug) {
+      System.out.println("Connecting to the masters at " + masterQuorum);
+    }
+
+    List<ColumnSchema> columns = new ArrayList<ColumnSchema>(11);
```
Comment (cmccoy) [inline kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java]: There's no guarantee that there will be 10 non-key columns. Can you make this a variable based on the value of `fieldcount`?

File: kudu/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Use the following command line to load the initial data into an existing Kudu cluster with default
+configurations.
+
+```
+bin/ycsb load kudu -P workloads/workloada
+```
+
+Additional configurations:
+* `masterQuorum`: The master's address. The default configuration expects a master on localhost.
```
Comment (cmccoy) [inline kudu/README.md]: Consider prefixing these with `kudu.`

File: kudu/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+bin/ycsb load kudu -P workloads/workloada
+```
+
+Additional configurations:
+* `kudu_master_addresses`: The master's address. The default configuration expects a master on localhost.
+* `kudu_pre_split_num_tablets`: The number of tablets (or partitions) to create for the table. The default
+uses 4 tablets. A good rule of thumb is to use 5 per tablet server.
+* `kudu_table_num_replicas`: The number of replicas that each tablet will have. The default is 3. Should
+only be configured to use 1 instead, for single node tests.
+* `kudu_sync_ops`: If the client should buffer data before sending it. The default is false. Should
```
Comment (cmccoy) [inline kudu/README.md]: If this should always be true for the run phase, maybe that should be the default?

**Comment by cmccoy** [comment]
Thanks! One more question, otherwise looks good.

File: kudu/README.md (hunk 3 above)
Comment (busbey) [inline kudu/README.md]: we ask datastore bindings to default to no client-side buffering. I'd rather have this rephrased to not suggest it always be true for the run phase. we do the same to e.g. HBase, Cassandra, etc.

Comment (cmccoy) [inline kudu/README.md]: IIUC, `true` indicates that there *shouldn't* be client side buffering, so having it be true for the run phase is correct. I think the default should be true as well.

Comment (jdcryans) [inline kudu/README.md]: Ah yeah my description here is off. True means it won't buffer. I'll make it default to that and change that line.

**Comment by cmccoy** [comment]
Thanks, @jdcryans!

**Pull request merged**